source/distributed/tria.cc

```diff
@@ -2550,6 +3194,128 @@ namespace parallel
       update_cell_relations();
     }
 
+    template <int dim, int spacedim>
+    unsigned int
+    Triangulation<dim, spacedim>::find_point_owner_rank(const Point<dim> &p)
+    {
+      // Call the other function
+      std::vector<Point<dim>>   point(1, p);
+      std::vector<unsigned int> owner = find_point_owner_rank(point);
+
+      return owner[0];
+    }
+
+    template <int dim, int spacedim>
+    std::vector<unsigned int>
+    Triangulation<dim, spacedim>::find_point_owner_rank(
+      const std::vector<Point<dim>> &
+#  ifdef DEAL_II_WITH_P4EST_SEARCH_LOCAL
+        // We need to do this, otherwise the compiler complains about an unused
+        // variable if the define is not present.
+        points
+#  endif
+    )
+    {
+#  ifndef DEAL_II_WITH_P4EST_SEARCH_LOCAL
+      AssertThrow(
+        false,
+        ExcMessage(
+          "This function is only available if p4est is version 2.2 and higher."));
+      // Just return to satisfy compiler
+      return std::vector<unsigned int>(1,
+                                       dealii::numbers::invalid_unsigned_int);
+#  else
+      // We can only use this function if vertices are communicated to p4est
+      AssertThrow(this->is_vertices_to_p4est_communicated(),
+                  ExcMessage(
+                    "Vertices need to be communicated to p4est to use this "
+                    "function. This must explicitely be turned on in the "
+                    "settings of the triangulation's constructor."));
+
+      // We can only use this function if all manifolds are flat
+      const std::vector<types::manifold_id> all_manifold_ids =
+        this->get_manifold_ids();
+      for (const auto &manifold_id : all_manifold_ids)
+        {
+          AssertThrow(
+            manifold_id == numbers::flat_manifold_id,
+            ExcMessage(
+              "Thsi function can only be used if the triangulation "
+              "has no other manifold than a Cartesian (flat) manifold attached."));
+        }
+
+      // Create object for callback
+      PartitionSearch<dim> partition_search;
+
+      // Pointer should be this triangualtion before we set it to something else
+      Assert(parallel_forest->user_pointer == this, ExcInternalError());
+
+      // re-assigne p4est's user pointer
+      parallel_forest->user_pointer = &partition_search;
+
+      /*
+       * ********************************************
+       * Copy points into p4est internal array data struct
+       */
```

Comment:
```suggestion
       //
       // Copy points into p4est internal array data struct
       //
```
We usually reserve this style of commenting for documentation in the header files. Are you okay with this?